elastalert/schema.yaml

```diff
@@ -25,7 +25,7 @@ definitions:
 
   filter: &filter {}
 
-required: [type, index, alert]
```

Comment:
why this change? It will be required for log alert rules?

Reply:
its validating every rule with this schema first hand, if we dont remove this, we ll have to pass atleast a dummy index in each rule config.

Reply:
we will add a default index in config.yaml, which will be added to each rule when index not specified.

Reply:
In that case, why would this be required then?

Reply:
This is needed because, the flow goes like
-> validates each rule with **schema.yaml** file ( so if we don't give index in rule - it ll throw error, thats why removed from schema)
-> Adds all the configs in **config.yaml** to all rules if its not present in individual rule config.

previously it was like, index field is a must in every indicidual rule config. as i removed index from required of schema.yaml, it can also be set in main config.yaml now.